rust/ql/src/queries/unusedentities/UnusedVariable.qll

```diff
@@ -1,18 +1,28 @@
 import rust
 
-/** A deliberately unused variable. */
+/**
+ * A deliberately unused variable, for example `_` or `_x`.
+ */
 class DiscardVariable extends Variable {
   DiscardVariable() { this.getName().charAt(0) = "_" }
 }
 
-/** Holds if variable `v` is unused. */
+/**
+ * Holds if variable `v` is unused.
+ */
 predicate isUnused(Variable v) {
   // variable is accessed or initialized
   not exists(v.getAnAccess()) and
-  not exists(v.getInitializer()) and
-  // variable is intentionally unused
-  not v instanceof DiscardVariable and
-  // variable is in a context where is may not have a use
-  not v.getPat().isInMacroExpansion() and
-  not exists(FnPtrType fp | fp.getParamList().getParam(_).getPat() = v.getPat())
+  not exists(v.getInitializer())
```

Comment:
@hvitved Should there be a "write access" for a variable with an initializer? The following two examples should probably be considered equivalent.
```
let x : i32 = 10;
```
and 
```
let x : i32;
x = 10;
```

Reply:
The `x` in `let x : i 32 = 10;` is a pattern and not an expression, so making it a write access is not that natural.